### ISSUE TITLE
wc-e2e does not use the latest WordPress version by default

### ISSUE THREAD

**Issue by ecgan** [woocommerce/woocommerce#30115]
**Prerequisites (mark completed items with an [x]):**
- [x] I have have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

**Describe the bug**

In [WooCommerce End to End Tests](https://github.com/woocommerce/woocommerce/tree/trunk/tests/e2e#running-tests), the page says: 

> Note that by default, Docker will download the latest images available for WordPress, PHP and MariaDB.

When I run `npx wc-e2e docker:up` without specifying `WP_VERSION`, I end up getting `wordpress` image with version 5.5.1. 

![image](https://user-images.githubusercontent.com/417342/122266683-a2a1c500-cf0c-11eb-9ad9-517a4a96f2a4.png)

It seems like this is related to:

https://github.com/woocommerce/woocommerce/blob/f3c9be0492140f9e99a4dadece4cdb66ce5a0a79/tests/e2e/env/bin/docker-compose.sh#L13

Just a few lines above it, I see a call to `get-latest-docker-tag.js`, so I manually run it in my terminal:

```text
woocommerce % ./tests/e2e/env/bin/get-latest-docker-tag.js wordpress 5
TypeError: Cannot read property 'last_updated' of null
    at paginationFn (/Users/engchin/Code/woocommerce/woocommerce/tests/e2e/env/bin/get-latest-docker-tag.js:90:51)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

The error is thrown from here:

https://github.com/woocommerce/woocommerce/blob/f3c9be0492140f9e99a4dadece4cdb66ce5a0a79/tests/e2e/env/bin/get-latest-docker-tag.js#L90

**Expected behavior**

The `wordpress` docker image should be the latest version.

**Actual behavior**

The `wordpress` docker image is version 5.5.1, not the latest version.

**Steps to reproduce the bug (We need to be able to reproduce the bug in order to fix it.)**

1. Make sure docker is started in your local machine.
2. Go to terminal. Make sure you have Node LTS version.
3. Run `npm install @woocommerce/e2e-environment` in your project.
4. Run `npx wc-e2e docker:up` (Make sure WP_VERSION is not set.)
5. When the process is done, go to Docker Dashboard and check the WordPress image version.

**Isolating the problem (mark completed items with an [x]):**

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**WordPress Environment**

Not applicable.

**Comment by adrianduffell** [comment]
We've started seeing this in the WC Admin e2e tests as well.

**Comment by zhongruige** [comment]
Just noting here that the same issue also applies to getting the PHP version in the same script:

https://github.com/woocommerce/woocommerce/blob/f3c9be0492140f9e99a4dadece4cdb66ce5a0a79/tests/e2e/env/bin/docker-compose.sh#L17

```
./tests/e2e/env/bin/get-latest-docker-tag.js php 7    
TypeError: Cannot read property 'last_updated' of null
    at paginationFn (/Users/zhongruige/workspace/woocommerce/tests/e2e/env/bin/get-latest-docker-tag.js:98:51)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

The MariaDB one, however, works as expected:

```
./tests/e2e/env/bin/get-latest-docker-tag.js mariadb 10
10.6.1
```

**Comment by rrennick** [comment]
This issue has been intermittently occurring because the `last_updated` field WP 5.8.0 tag in docker hub api has been bumped several times since WP 5.8 was released. In the past, once released that timestamp was not updated. 

The E2E package has a built in 12 hour delay to accommodate the api being updated with a new release before the archive is available for download.

Closing this as https://github.com/woocommerce/woocommerce/pull/30529 bumped the fallback versions to the current available versions.

**Comment by ecgan** [comment]
@rrennick , the fix implemented in #30529 is done by changing the hardcoded version 5.5.1 to hardcoded version 5.8.0. When future version of WP is released (say 5.9.0), and if it's not updated in the wc-e2e package again, then this issue #30115 would happen again, no? 

I'm wondering if there is a way to make wc-e2e to always get the latest version without hardcoding the version number? 🤔  Would [`wordpress:latest`](https://hub.docker.com/layers/wordpress/library/wordpress/latest/images/sha256-14f9c6683815aa90466a1f2e057aa9312989e19cb35d26da0a4db0e07fb1e875?context=explore) work?

**Comment by rrennick** [comment]
> Would `wordpress:latest` work?

No, it doesn't. If you use `wordpress:latest` in your `docker-compose.yml` then you need to run `docker pull` before each start. When `latest` is used in a DC file, Docker internally converts that to the numerical version the first time the container is started. Subsequent starts use the same internal numerical version.

> if it's not updated in the wc-e2e package again, then this issue #30115 would happen again, no?

Yes. WC core already has processes in place to bump versions (eg. unit test matrix). We didn't have this one on the list to be bumped. We'll address this before the next major WP release.
